src/Storages/System/StorageSystemUsers.cpp

```diff
@@ -86,7 +89,18 @@ void StorageSystemUsers::fillData(MutableColumns & res_columns, const Context &
         column_id.push_back(id);
         column_storage.insertData(storage_name.data(), storage_name.length());
         column_auth_type.push_back(static_cast<Int8>(authentication.getType()));
-        column_auth_params_offsets.push_back(column_auth_params.size());
+
+        if (authentication.getType() == Authentication::Type::LDAP_SERVER)
+        {
+            Poco::JSON::Object auth_params_json;
+            auth_params_json.set("server", authentication.getServerName());
+
+            std::ostringstream oss;
+            Poco::JSON::Stringifier::stringify(auth_params_json, oss);
+            const auto str = oss.str();
+
+            column_auth_params.insertData(str.data(), str.size());
```

Comment:
This is a table, the number of rows in each column should be the same.
So it's necessary to add something to the column `column_auth_params` anyway.

Reply:
That makes sense. I'll write `{}` for general case, to kind of hint that a JSON object is what we store there. This should probably also fix a regression detected by integration test I have right now.

Reply:
Done. Thanks.

Reply:
For authentication type other than `LDAP_SERVER` the following code

```
std::string_view empty_json = "{}";
column_auth_params.insertData(empty_json.data(), empty_json.size());
```

works much faster than your code.

Reply:
Yes, obviously. But does it worth it? Adding other new cases there will be kind of less intrusive.

Reply:
Will change that, but I believe such special cases only add entropy to the code. (And this looks like not a perf-sensitive code.)

Reply:
Done.